### PR TITLE
Re-implement Module, Environment, Binding Helper in C

### PR DESCRIPTION
**Pull request by yichoi**
IoT.js-DCO-1.0-Signed-off-by: Youngil Choi duddlf.choi@samsung.com

### CODE REVIEW

File: src/platform/iotjs_module_gpio-linux-general.inl.h [hunk 1]
```diff
@@ -86,13 +86,13 @@ class LocalFsReq : public uv_fs_t {
 
 // Checks if given directory exits.
```
Comment (chokobole) [inline src/platform/iotjs_module_gpio-linux-general.inl.h]: exists?

File: src/iotjs_env.c [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+#include <string.h>
+
+
+static iotjs_environment_t env;
+static bool initialized = false;
+
+
+/**
+* Constructor/Destructor on private section.
```
Comment (zherczeg) [inline src/iotjs_env.c]: I think a space is missing before *.

File: src/iotjs_env.h [hunk 3]
```diff
@@ -18,109 +18,65 @@
...
 
   // Running state.
-  State _state;
+  State state;
 
   // Run config
-  Config _config;
+  Config config;
+
+} IOTJS_VALIDATED_STRUCT(iotjs_environment_t);
```
Comment (zherczeg) [inline src/iotjs_env.h]: This newline is not needed

**Comment by zherczeg** [comment]
Very nice patch. LGTM after some minor style fixes.

**Comment by yichoi** [comment]
Rebased and updated the commit as @zherczeg commented.

**Comment by nova0821** [comment]
LGTM. Great patch!

**Pull request merged**